src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -773,6 +832,20 @@ public GetRecordsResult getRecords(GetRecordsRequest request) {
         });
     }
 
+    private boolean acquirePermit(StreamShardPosition position) {
+        final boolean[] acquired = new boolean[] { false };
+        try {
+            permitCache.get(position, () -> {
```

Comment:
I guess this is here so as to avoid hitting the rate limiter unnecessarily?

Reply:
The cache itself itself is the rate limiter. The cache is configured to evict each entry after a certain period of time (by default 100ms). If `get` calls callable, it means there is no entry or the current entry is expired. If it doesn't call the callable, then there is a non-expired entry.